dspy/predict/predict.py

```diff
@@ -86,7 +94,7 @@ def forward(self, **kwargs):
         if self.lm is None:
             x, C = dsp.generate(signature, **config)(x, stage=self.stage)
         else:
-            with dsp.settings.context(lm=self.lm, query_only=True):
```

Comment:
Ok but we'll need to query_only=True somewhere... I guess we can have `self.query_only` in the module itself... Basically this is affecting now the finetune notebook in the examples/qa/hotpot folder

Reply:
When is query_only used? Query_only seems it should be a kwarg for the forward method. I'm not sure where you'd want your model to not send demonstrations when they are set.

Reply:
Need a solve.